src/hooks/chart.ts

```diff
@@ -269,11 +279,26 @@ export const useChartData = (plot: Plot) => {
     isFetchSelected,
     setIsFetchSelected,
   } = useChartSelectGrid({
+    plot,
     templateParams,
     setModalProps,
     setIsModalOpen,
   });
 
+  useEffect(() => {
```

Comment:
Add a comment explaining when this will trigger since it is updating the controls. 

Should this be updated each time `templateParams` is changed? If we have controls that rely on each other, then maybe this makes sense. But I think we should be initializing the selectors once instead of on a change. Does this also mean the selector data is updated when the "plot" changes? (when any of the layout, config, or data changes...?)

Reply:
Yes it should be initialized once only.

Reply:
Please add a comment to this useEffect then stating this should trigger only once as setup